### PR TITLE
fix: DefinedRouteCollector to use RouteCollectionInterface

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
Supersedes #8206
See #6814

- fix RouteCollectionInterface
- DefinedRouteCollector uses RouteCollectionInterface instead of RouteCollection

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Review by datamweb — COMMENTED** [review]
Looks good so far.

**Review by michalsn — APPROVED** [review]
Ok, we just need some changelog.

File: phpstan-baseline.php [hunk 1]
```diff
@@ -7197,34 +7197,9 @@
...
-$ignoreErrors[] = [
-	'message' => '#^Call to an undefined method CodeIgniter\\\\Router\\\\RouteCollectionInterface\\:\\:isFiltered\\(\\)\\.$#',
-	'count' => 1,
-	'path' => __DIR__ . '/system/Router/Router.php',
-];
-$ignoreErrors[] = [
-	'message' => '#^Call to an undefined method CodeIgniter\\\\Router\\\\RouteCollectionInterface\\:\\:setHTTPVerb\\(\\)\\.$#',
-	'count' => 1,
-	'path' => __DIR__ . '/system/Router/Router.php',
+    'message' => '#^Call to an undefined method CodeIgniter\\\\Router\\\\RouteCollectionInterface\\:\\:getRegisteredControllers\\(\\)\\.$#',
```
[the remark this answers is not in the document]
Comment (kenjis) [reply phpstan-baseline.php]: The method is only for AutoRouting Improved, and marked as `@interal`.
I don't think the RouteCollectionInterface should have it.

**Comment by kenjis** [comment]
Added docs.

**Review by paulbalandan — APPROVED** [review]

**Pull request merged**